python/lsst/pipe/base/executionButlerBuilder.py

```diff
@@ -227,7 +227,31 @@ def _export(
     return yamlBuffer
 
 
-def _setupNewButler(butler: Butler, outputLocation: ResourcePath, dirExists: bool) -> Butler:
+def _setupNewButler(
+    butler: Butler,
+    outputLocation: ResourcePath,
+    dirExists: bool,
+    datastoreRoot: Optional[str] = None,
+) -> Butler:
+    """Set up the execution butler
+
+    Parameters
+    ----------
+    butler : `Butler`
+        The original butler, upon which the execution butler is based.
+    outputLocation : `ResourcePath`
+        Location of the execution butler.
+    dirExists : `bool`
+        Does the ``outputLocation`` exist, and if so, should it be clobbered?
+    datastoreRoot : `str`, optional
```

Comment:
I guess we should probably support `ResourcePath` here as well and then do `str(datastoreRoot)` below.